python/huggingfaceserver/huggingfaceserver/utils.py

```diff
@@ -13,19 +13,25 @@
 # limitations under the License.
 
 import torch
+import os
 from transformers import PretrainedConfig
-from typing import Optional
+from typing import Optional, Union, List, Any
 
 from kserve.logging import logger
 
 
+ALLOW_LONG_MAX_MODEL_LEN = "ALLOW_LONG_MAX_MODEL_LEN"
+
+
 # "This implementation is based on vLLM's _get_and_verify_max_len "
-# "https://github.com/vllm-project/vllm/blob/a377f0bd5e1fa0ca069e3dbf28f4de5af64d0bb1/vllm/config.py#L1160"
+# "https://github.com/vllm-project/vllm/blob/2ac6d0e75bc846998da56b50bf4f8853cb36d484/vllm/config.py#L1870"
```

Comment:
I believe you can consume these methods through the vLLM pkg now

Reply:
I have the same question if we can reuse this function from vllm ?

Reply:
As of now, we don't mandate installation/existence of vLLM package. Its an extra and if one chooses to run on CPU or just via HuggingfaceGenerativeModel, it'd be an unnecessary.
Do we want to make vLLM mandataory?